### PR TITLE
Update with abstracted network and improve patches

### PR DESCRIPTION
**Pull request by natereid72**
### Description of your changes

Moved network to nested XR of Cluster composition and removed need for NetworkID. Removed or modified some Composition patches for not being needed and/or improved readability. Update Readme for redesigned provisioning flow.

I have:

- [x ] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x ] Run `make reviewable` to ensure this PR is ready for review.
- [x ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

Performed testing on self-hosted and upbound hosted control planes.

### CODE REVIEW

**Comment by natereid72** [comment]
The cluster deploy from UI needs some adjusting. Working on that today and will update PR.

**Comment by natereid72** [comment]
The cluster deploy frrom UI needs some fixing, will update PR and comment today

File: README.md [hunk 1]
```diff
@@ -216,24 +216,10 @@ Crossplane resources use the ProviderConfig named ```default``` if no specific P
 
 ### We are now ready to provision resources:
 
-#### Create Network Fabric
```
Comment (jbw976) [inline README.md:219]: we should also update the "APIs in this Configuration" section that still lists the Network type

File: README.md [hunk 2]
```diff
@@ -317,7 +307,9 @@ az ad sp delete --id $AZ_APP_ID
 
 Create a `Repository` called `platform-ref-azure` in your Upbound Cloud `Organization`:
 
-![Upbound Repository](docs/media/repository.png)
+![](docs/media/2022-01-12-15-56-26.png)
```
Comment (jbw976) [inline README.md:310]: can we give these media files more descriptive file names, instead of timestamps? e.g. `01-repository-empty.png` or similar

Comment (jbw976) [reply README.md:310]: also, are all the new media files used? i only see 2 of them referenced during a quick scan

Comment (natereid72) [reply README.md:315]: > also, are all the new media files used? i only see 2 of them referenced during a quick scan

They are not, they were used in a different, non-committed iteration. I will delete unused.

Comment (natereid72) [reply README.md:315]: > can we give these media files more descriptive file names, instead of timestamps? e.g. `01-repository-empty.png` or similar

I will, I need to reconfigure my vsc extension that auto-names those.

Comment (natereid72) [reply README.md]: I believe it's good now.

File: README.md (hunk 1 above)
Comment (natereid72) [reply README.md:219]: I believe it's good now.

**Comment by natereid72** [comment]
@elohmrow  found a blocker with the order of installing `up` cli and submitting the `up list ctp` command in the readme. He is going to open an issue and we will resolve.

**Comment by jbw976** [comment]
I took a test pass through this and I also see what @ezgidemirel was referring to this morning where not everything makes it to the READY status, because the XAKS object has a `providerconfig.helm.crossplane.io` that has no concept of a READY condition.

Looking at the composition docs in https://crossplane.io/docs/v1.6/reference/composition.html#readiness-checks (and other spots on that page), it looks like we could update the XAKS composition to only consider the ready condition of its `AKSCluster` resource and ignore the `providerconfig.helm` one.

See if it helps to set NO readiness check on the `providerconfig.helm` resource, as described in the below snippet from the docs:

```yaml
    # By default an XR's 'Ready' status condition will become True when the
    # 'Ready' status conditions of all of its composed resources become true.
    # You can optionally specify custom readiness checks to override this.
    readinessChecks:
    - type: None
```

File: cluster/services/composition.yaml [hunk 3]
```diff
@@ -22,9 +22,15 @@ spec:
...
-              version: "15.0.1"
+              version: "15.0.2"
             values: {}
       patches:
+        - fromFieldPath: spec.id
+          toFieldPath: metadata.name
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-prometheus"
```
Comment (jbw976) [inline cluster/services/composition.yaml:33]: i'm not sure if this patch worked as intended, this is the release.helm.crossplane.io that i had when testing just now (note that I don't see `*-prometheus` in its name):
```
> k get managed
NAME                                                              CHART        VERSION   SYNCED   READY   STATE      REVISION   DESCRIPTION        AGE
release.helm.crossplane.io/platform-ref-azure-jared-n6rsv-lch9x   prometheus   15.0.1    True     True    deployed   1          Install complete   25m
```

Comment (jbw976) [reply cluster/services/composition.yaml:33]: The general patching naming approach did seem to work OK though for other MRs (note `*-rg`, `*-postgresql`, `*-sn`, etc. for example in other names)
```yaml
> k get managed
NAME                                                              CHART        VERSION   SYNCED   READY   STATE      REVISION   DESCRIPTION        AGE
release.helm.crossplane.io/platform-ref-azure-jared-n6rsv-lch9x   prometheus   15.0.1    True     True    deployed   1          Install complete   25m

NAME                                                            READY   SYNCED
resourcegroup.azure.crossplane.io/platform-ref-azure-jared-rg   True    True

NAME                                                                  READY   SYNCED   ENDPOINT                                                  LOCATION    AGE
akscluster.compute.azure.crossplane.io/platform-ref-azure-jared-aks   True    True     platform-ref-azure-jared-7977a8a3.hcp.westus2.azmk8s.io   West US 2   25m

NAME                                                                                READY   SYNCED   VERSION   AGE
postgresqlserver.database.azure.crossplane.io/platform-ref-azure-jared-postgresql   False   False    11        3m42s

NAME                                                                                               READY   SYNCED   STATE   AGE
postgresqlservervirtualnetworkrule.database.azure.crossplane.io/platform-ref-azure-jared-sn-rule   False   False            3m42s

NAME                                                             READY   SYNCED   STATE       AGE
subnet.network.azure.crossplane.io/platform-ref-azure-jared-sn   True    True     Succeeded   25m

NAME                                                                       READY   SYNCED   STATE       LOCATION    AGE
virtualnetwork.network.azure.crossplane.io/platform-ref-azure-jared-vnet   True    True     Succeeded   West US 2   25m
```

**Review by jbw976 — APPROVED** [review]
This is looking pretty good @natereid72, I think this is a definite experience improvement where users don't have to worry about the infra details of Networking, that's abstracted away now from the app developer.

A couple comments and things to look into, but otherwise LGTM and adding my approval now.

**Comment by jbw976** [comment]
Ah, some other feedback I forgot:

* I was going to suggest to bump the version number in the readme to v0.2.0, but we haven't done a v0.1.0 release yet still.  Let's move ahead after getting this PR in to do a release and publish it and make a bit of noise :)
* I'm starting to feel like we should streamline these readmes across all reference platforms - they have a lot of duplication, they have somewhat old messaging, they are hard to follow with all the instructions, etc.  Can you open an issue for us to take a pass at the readmes and try to streamline them?
  * I can follow along with it because my trained eyes know which steps are important and what can be glossed over.  Definitely a sign that we should streamline so newer users don't have to parse through so much 🤪

**Comment by natereid72** [comment]
Service name suffix fixed
Object readiness fixed

**Comment by ezgidemirel** [comment]
> I took a test pass through this and I also see what @ezgidemirel was referring to this morning where not everything makes it to the READY status, because the XAKS object has a `providerconfig.helm.crossplane.io` that has no concept of a READY condition.
> 
> Looking at the composition docs in https://crossplane.io/docs/v1.6/reference/composition.html#readiness-checks (and other spots on that page), it looks like we could update the XAKS composition to only consider the ready condition of its `AKSCluster` resource and ignore the `providerconfig.helm` one.
> 
> See if it helps to set NO readiness check on the `providerconfig.helm` resource, as described in the below snippet from the docs:
> 
> ```yaml
>     # By default an XR's 'Ready' status condition will become True when the
>     # 'Ready' status conditions of all of its composed resources become true.
>     # You can optionally specify custom readiness checks to override this.
>     readinessChecks:
>     - type: None
> ```

@jbw976 I mentioned this in purple squad's stand up, and colleagues shared some history. Looks like, it's a known [issue](https://github.com/crossplane/crossplane/issues/1687) discussed and [documented](https://crossplane.io/docs/v1.6/reference/composition.html) already.

**Comment by natereid72** [comment]
It tested out with:
```
    readinessChecks:
    - type: None
```

But I ended up going with:
```
    readinessChecks:
    - type: MatchString
    fieldPath: status.atProvider.state
    matchString: "deployed"
```

**Comment by natereid72** [comment]
I think it was @jbw976 that mentioned XR not ready while MR was ready. I'm seeing that today and haven't figured it out yet
image

**Comment by jbw976** [comment]
@natereid72 I would imagine that is from the readinessCheck that you have now added for the helm providerConfig - i think the match you added of `status.atProvider.state == "deployed"` doesn't make sense because the helm providerConfig doesn't have a status field like that.  Did we mix up the helm providerConfig and the helm release objects?

I think the XAKS composition should probably just use a readiness check of None for the helm providerConfig.

**Comment by natereid72** [comment]
Yeah, that makes sense. I think I had that and then thought everything was fine with the change. I'll patch that up. Thanks for the look.

**Comment by natereid72** [comment]
@jbw976 Ok, looks good now. The placement of the release check was in the actual services and then I moved it for some reason I can't understand now (I'm guessing too early in the morning pebcak). Pushing update now.

**Comment by natereid72** [comment]
big screen of `ready`

image

**Comment by jbw976** [comment]
That's a lot of READY=true @natereid72.  Shall we go ahead and merge?
